File: frontend/src/services/NoteService.js
import ApiService from "./base/ApiService";




const NOTE_URL = '/note'
class UserService {

    get(url) {
        return ApiService.get(url)
    }
    search(body) {
        return ApiService.post(NOTE_URL + "/search", body)
    }
    save(body) {
        return ApiService.post(NOTE_URL + "/save", body)
    }
    update(body) {
        return ApiService.put(NOTE_URL + "/update", body)
    }
    delete(body) {
        return ApiService.post(NOTE_URL + "/delete", body)
    }
    findById(body) {
        return ApiService.post(NOTE_URL + "/findById", body)
    }

    getNotesWithPagination(page, size) {
        let url = '/search?page=' + page + '&size=' + size;
        return ApiService.get(NOTE_URL + url);
    }
}

export default new UserService();

File: backend/src/main/java/com/project/amaris/note/service/NoteService.java
package com.project.amaris.note.service;

import com.project.amaris.note.dto.NoteDto;
import com.project.amaris.note.entity.Note;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface NoteService {


    public List<Note> findAll();
    public Page<NoteDto> search(Pageable page) throws Exception;
    public Boolean save(Note note) throws Exception;
    public NoteDto findById(Note note) throws Exception;
    public Boolean update(Note note) throws Exception;
    public Boolean delete(Note note) throws Exception;
}
